src/serializer/LazyObjectsSerializer.js

```diff
@@ -105,6 +118,121 @@ export class LazyObjectsSerializer extends ResidualHeapSerializer {
     return getOrDefault(this._valueLazyIds, obj, () => this._lazyObjectIdSeed++);
   }
 
+  _isLazyObject(val: Value) {
+    // Currently, only support lazy raw object.
+    return this._isRawObject(val);
+  }
+
+  // Must keep in-sync with ResidualHeapSerializer logic.
+  _isObject(val: Value) {
+    return !(
+      val instanceof AbstractValue ||
+      val.isIntrinsic() ||
+      val instanceof EmptyValue ||
+      val instanceof UndefinedValue ||
+      ResidualHeapInspector.isLeaf(val) ||
+      IsArray(this.realm, val) ||
+      val instanceof ProxyValue ||
+      val instanceof FunctionValue ||
+      val instanceof SymbolValue
+    );
+  }
+
+  // Must keep in-sync with ResidualHeapSerializer logic.
```

Comment:
In sync with what exactly?

Reply:
In sync with ResidualHeapSerializer.serializeValueObject() method logic. Will add.